lib/trino-filesystem/src/test/java/io/trino/filesystem/AbstractTestTrinoFileSystem.java

```diff
@@ -0,0 +1,837 @@
+    {
+        // rename file locations cannot be the root of the file system
+        assertThatThrownBy(() -> getFileSystem().renameFile(getRootLocation(), createLocation("file")))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining(getRootLocation());
+        assertThatThrownBy(() -> getFileSystem().renameFile(createLocation("file"), getRootLocation()))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining(getRootLocation());
+        assertThatThrownBy(() -> getFileSystem().renameFile(getRootLocation() + "/", createLocation("file")))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining(getRootLocation() + "/");
+        assertThatThrownBy(() -> getFileSystem().renameFile(createLocation("file"), getRootLocation() + "/"))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining(getRootLocation() + "/");
+        // rename file locations cannot end with a slash
+        assertThatThrownBy(() -> getFileSystem().renameFile(createLocation("foo/"), createLocation("file")))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining(createLocation("foo/"));
+        assertThatThrownBy(() -> getFileSystem().renameFile(createLocation("file"), createLocation("foo/")))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining(createLocation("foo/"));
+        // rename file locations cannot end with whitespace
+        assertThatThrownBy(() -> getFileSystem().renameFile(createLocation("foo "), createLocation("file")))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining(createLocation("foo "));
+        assertThatThrownBy(() -> getFileSystem().renameFile(createLocation("file"), createLocation("foo ")))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining(createLocation("foo "));
+        assertThatThrownBy(() -> getFileSystem().renameFile(createLocation("foo\t"), createLocation("file")))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining(createLocation("foo\t"));
+        assertThatThrownBy(() -> getFileSystem().renameFile(createLocation("file"), createLocation("foo\t")))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining(createLocation("foo\t"));
+
+        // todo rename to existing file name
+        try (TempBlob sourceBlob = randomBlobLocation("renameSource");
+                TempBlob targetBlob = randomBlobLocation("renameTarget")) {
+            // renaming a non-existent file is an error
+            assertThatThrownBy(() -> getFileSystem().renameFile(sourceBlob.location(), targetBlob.location()))
+                    .isInstanceOf(IOException.class)
```

Comment:
Should this be `java.io.FileNotFoundException` or `java.nio.file.NoSuchFileException`?